### PR TITLE
ScalaDoc for Mux (examples added)

### PR DESCRIPTION
**Pull request by schoeberl**
as discussed in the meeting

### CODE REVIEW

File: src/main/scala/chisel3/util/Mux.scala [hunk 1]
```diff
@@ -10,6 +10,15 @@ import chisel3.core.SeqUtils
 
 /** Builds a Mux tree out of the input signal vector using a one hot encoded
   * select signal. Returns the output of the Mux tree.
+  * 
+  * @example {{{
+  * val hotValue = chisel3.util.oneHotMux(Seq(
```
Comment (edwardcwang) [inline src/main/scala/chisel3/util/Mux.scala]:
```suggestion
  * val hotValue = chisel3.util.Mux1H(Seq(
```

File: src/main/scala/chisel3/util/Mux.scala [hunk 2]
```diff
@@ -33,7 +50,13 @@ object PriorityMux {
   def apply[T <: Data](sel: Bits, in: Seq[T]): T = apply((0 until in.size).map(sel(_)), in)
 }
 
-/** Creates a cascade of n Muxs to search for a key value. */
+/** Creates a cascade of n Muxs to search for a key value.
+  *
+  * @example {{{
+  * MuxLookup(idx, default, 
+  *     Array(0.U -> a, 1.U -> b, ...))
```
Comment (edwardcwang) [inline src/main/scala/chisel3/util/Mux.scala]:
```suggestion
  *     Array(0.U -> a, 1.U -> b))   
```

File: src/main/scala/chisel3/util/Mux.scala [hunk 3]
```diff
@@ -50,6 +73,11 @@ object MuxLookup {
 
 /** Given an association of values to enable signals, returns the first value with an associated
   * high enable signal.
+  *
+  * @example {{{
+  * MuxCase(default, Array(c1 -> a, c2 -> b, ...))
```
Comment (edwardcwang) [inline src/main/scala/chisel3/util/Mux.scala]:
```suggestion
  * MuxCase(default, Array(c1 -> a, c2 -> b))
```

File: src/main/scala/chisel3/util/Mux.scala (hunk 1 above)
Comment (edwardcwang) [reply src/main/scala/chisel3/util/Mux.scala]: So what is the relationship between `oneHotMux` and `Mux1H`?

**Comment by schoeberl** [comment]
I am not sure what I have to do now. GitHub says "Changes requested", but I thought I've merged all commits from @edwardcwang into this PR.

Martin

**Comment by schoeberl** [comment]
I don't have write access, so I cannot merge ;-)

**Comment by edwardcwang** [comment]
retest this please

**Pull request merged**